manual/src/refman/extensions/alerts.etex

```diff
@@ -17,31 +17,68 @@ component is referenced).
 
 The "ocaml.alert" or "alert" attribute serves two purposes: (i) to
 mark component with an alert to be triggered when the component is
-referenced, and (ii) to control which alert names are enabled.  In the
-first form, the attribute takes an identifier possibly
-followed by a message. Here is an example of a value declaration marked
-with an alert:
+referenced, and (ii) to control which alert names are enabled.
 
-\begin{verbatim}
+\subsection{ss:alert_marking}{Marking a component}
+
+When a marking a component, the attribute attribute takes an identifier
+possibly followed by a message. Here is an example of a value declaration
+marked with an alert:
+
+\begin{caml_example*}{signature}
 module U: sig
   val fork: unit -> bool
     [@@alert unix "This function is only available under Unix."]
 end
-\end{verbatim}
+\end{caml_example*}
 
 Here "unix" is the identifier for the alert.  If this alert category
 is enabled, any reference to "U.fork" will produce a message at
 compile time, which can be turned or not into a fatal error.
 
-And here is another example as a floating attribute on top
-of an ``.mli'' file (i.e. before any other non-attribute item)
-or on top of an ``.ml'' file without a corresponding interface file,
-so that any reference to that unit will trigger the alert:
+For other structure or signature components, the syntax is as follows:
 
-\begin{verbatim}
+\begin{caml_example*}{verbatim}
+module type Sig = sig
+  val v: int [@@alert name "message"]
+end
+
+let v [@alert name "message"] = 1
+let `A (v [@alert name "message"]) = `A 1
+
+external e : int -> int = "c_function" [@@alert name "message"]
+
+type t1 = A | B [@@alert name "message"]
+(* N.B.: this only alerts on use of "t1", not uses of the constructors "A"
+   and "B". So let x = A wouldn't alert, so long as you don't write down its
+   type. *)
+
+type t2 = A | B [@alert name "message"]
+(* Alerts on use of "B", but not "A" nor "t2". Note the single "@". *)
+
+type t3 = { a : int; b : int [@alert name "message"] }
+(* Alerts on use of "b", but not "a" nor "t3". Note the single "@". *)
+
+exception E of int [@alert name "message"]
+(* Alerts on use of "E". Note the single "@". *)
+
+module M = struct end [@@alert name "message"]
+module type S = sig end [@@alert name "message"]
+class c1 = object end [@@alert name "message"]
+class type c2 = object end [@@alert name "message"]
+\end{caml_example*}
+
+Finally, to alert when the module corresponding to a file is referenced,
+a floating attribute can be written at the top of of the ``.mli'' file
+(i.e. before any other non-attribute item), or the top of the ``.ml'' file
+if there is no ``.mli'' file:
+
+\begin{caml_example*}{verbatim}
 [@@@alert unsafe "This module is unsafe!"]
-\end{verbatim}
+\end{caml_example*}
+
 
+\subsection{ss:control}{Controlling which alert names are enabled}
```

Comment:
`ss:alert_control`

Also I would say that it is the `alerts` that are enabled rather than their `names`:

> \subsection{ss:alert_control}{Controlling which alerts are enabled}